tests/test_corpus.py

```diff
@@ -201,6 +203,33 @@ def test_docfile_plain(tmpdir):
     assert len(list(docs.documents)) == 3
 
 
+def test_docfile_nonexistent(tmpdir):
+    docfile = tmpdir.join('documents_nonexistent.tsv')
+    with pytest.raises(AnnifException) as err:
+        docs = annif.corpus.DocumentFile(str(docfile))
+        list(docs.documents)
+    assert "No such file or directory: '{}'".format(docfile) in str(err.value)
+
+
+def test_docfile_plain_invalid_lines(tmpdir, caplog, capsys):
+    docfile = tmpdir.join('documents_invalid.tsv')
+    docfile.write("""Läntinen\t<http://www.yso.fi/onto/yso/p2557>
+
+        Oulunlinnan\t<http://www.yso.fi/onto/yso/p7346>
+        A line with no tabs
+        Harald Hirmuinen\t<http://www.yso.fi/onto/yso/p6479>""")
+    docs = annif.corpus.DocumentFile(str(docfile))
+    assert len(list(docs.documents)) == 3
+    # Need to capture both logs and stderr: when running this test individually
```

Comment:
Would it be cleaner to use [pytest-capturelog](https://pypi.org/project/pytest-capturelog/) to catch and check the log messages?

Reply:
The `capsys` was used because when full test suite was run, the line [`click_log.basic_config(logger)`](https://github.com/NatLibFi/Annif/blob/master/annif/cli.py#L21) made some changes to the logger, which in turn made the warnings intended to be captured in this test go to `stderr` and capturing them with `caplog` failed: 
```
------------------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------------------
warning: Skipping invalid line (missing tab): ""
warning: Skipping invalid line (missing tab): "        A line with no tabs"
```

The change due to that line in `logger.__dict__` is from 
`{'filters': [], 'name': 'annif', 'level': 0, 'parent': <RootLogger root (WARNING)>, 'propagate': True, 'handlers': [], 'disabled': False, 'manager': <logging.Manager object at 0x7fa66c0e9860>}`
to 
`{'filters': [], 'name': 'annif', 'level': 0, 'parent': <RootLogger root (WARNING)>, 'propagate': False, 'handlers': [<ClickHandler (NOTSET)>], 'disabled': False, 'manager': <logging.Manager object at 0x7fa66c0e9860>}`

By switching `'propagate': False` back to `True` makes the log capture [work in this test](https://github.com/NatLibFi/Annif/pull/299/commits/9041165f67de5a964b7004f1e1427ded63f7bf41). But I'm not so sure if it should be touched for this purpose, or would there be some way to test this?

Reply:
Discussion on [Handling of loggers with propagate=False](https://github.com/pytest-dev/pytest/issues/3697).